### PR TITLE
test/bad_dtls_test.c: remove dependency on SIXTY_FOUR_BIT_LONG.

### PR DESCRIPTION
**Pull request by dot-asm**

### CODE REVIEW

File: test/bad_dtls_test.c [hunk 1]
```diff
@@ -284,10 +284,11 @@ static int send_record(BIO *rbio, unsigned char type, unsigned long seqnr,
...
 
-#ifdef SIXTY_FOUR_BIT_LONG
-    seq[0] = (seqnr >> 40) & 0xff;
-    seq[1] = (seqnr >> 32) & 0xff;
-#endif
+    if (sizeof(seqnr) > 4) {
+        /* two-step shifts avoid shift-count-overflow warnings */
+        seq[0] = (seqnr >> 24 >> 16) & 0xff;
+        seq[1] = (seqnr >> 24 >> 8) & 0xff;
+    }
```
Comment (mattcaswell) [inline test/bad_dtls_test.c]: Wouldn't it be better to make seqnr a uint64_t?

Comment (dot-asm) [inline test/bad_dtls_test.c]: Like just pushed?

**Comment by mattcaswell** [comment]
Yes - but you missed a couple of spots. The "seq" field of the tests struct should be similarly changed to uint64_t on line 431...which then also implies similar changes to the recv_buf code lines 573-595

**Comment by dot-asm** [comment]
Well, it's not really like I missed those spots, as I did consider them, but decided to limit myself to send_record alone. Because doing otherwise felt like fixing some other problem completely. In other words I'd argue that if something needs to be done, it should be another request...

**Comment by mattcaswell** [comment]
I'd argue that you should have consistent types, so if you change one instance you should change them all. As it is now the seq field in the tests struct will have an implicit cast to uint64_t (which wasn't there before your change) for no good reason - so you are introducing the problem as a result of this change and should resolve it as part of this change. The same reasoning flows through to the recv_buf code. Its not like its a hard change :-).

**Comment by dot-asm** [comment]
> so you are introducing the problem as a result of this change

No, I'm not. As far as any externally measurable outcome goes, behaviour "before" and "after" is exactly same, *exactly*. Yes, there is implicit cast, but side effects are equivalent. Because the said cast results in zero-pad (on 32-bit platform) and initial values of first two elements of seq[6] are zeros.

**Comment by mattcaswell** [comment]
> > so you are introducing the problem as a result of this change
>
> No, I'm not

I disagree. My argument is that the implicit cast is a problem *in itself* which you are introducing. It adds unnecessary complexity to the code which we should avoid. The presence of a cast requires you to go through the thought process of reasoning that it is safe. That thought process will have to be repeated every time we change the code in this area to confirm that it is still safe. Therefore this represents an additional maintenance burden - which is entirely avoidable by a very small amount of additional work.

Sometimes casts are the only way to get the job done and are a necessary evil. This isn't one of those times.

**Comment by dot-asm** [comment]
> implicit cast is a problem in itself

Following this logic programming C is problem in itself. Have to go to re-evaluate my whole life...

**Pull request closed without merging**